### ISSUE TITLE
Lambda environment variables "API_XXXXXX_YYYYYYTABLE_NAME" is without "env" value.

### ISSUE THREAD

**Issue by chakimar** [aws-amplify/amplify-cli#3776]
**Describe the bug**
After `amplify function update` auto generated dynamodb variable without "env" value.

**Amplify CLI Version**
4.17.1

**To Reproduce**
```
$ amplify function update
Using service: Lambda, provided by: awscloudformation
? Please select the Lambda Function you would want to update 
function1
? Do you want to update permissions granted to this Lambda function to perform on other resources in your project? 
Yes
? Select the category 
storage
? Storage has 19 resources in this project. Select the one you would like your Lambda to access 
Foo:@model(appsync)
? Select the operations you want to permit for Foo:@model(appsync) 
create, read, update, delete
```

**Expected behavior**
```
console.log(process.env.ENV);
-> "dev"
console.log(process.env.API_XXXXXX_GRAPHQLAPIIDOUTPUT);
-> "xxxxxx"
console.log(process.env.API_XXXXXX_FOOTABLE_NAME)
-> "Foo-xxxxxx-dev"
```

function1-cloudformation-template.json
```
"Environment": {
	"Variables": {
		"ENV": {
			"Ref": "env"
		},
		"REGION": {
			"Ref": "AWS::Region"
		},
		"API_XXXXXX_FOOTABLE_NAME": {
			"Fn::ImportValue": {
				"Fn::Sub": "${apiXXXXXXGraphQLAPIIdOutput}:GetAtt:FooTable:Name"
			}
		}
	}
}
```

**Actual behavior**
```
console.log(process.env.ENV);
-> "dev"
console.log(process.env.API_XXXXXX_GRAPHQLAPIIDOUTPUT);
-> "xxxxxx"
console.log(process.env.API_XXXXXX_FOOTABLE_NAME)
-> "FooTable-xxxxxx"
```

function1-cloudformation-template.json
```
"Environment": {
	"Variables": {
		"ENV": {
			"Ref": "env"
		},
		"REGION": {
			"Ref": "AWS::Region"
		},
		"API_XXXXXX_FOOTABLE_NAME": {
			"Fn::Join": [
				"-",
				[
					"FooTable",
					{
						"Ref": "apixxxxxxGraphQLAPIIdOutput"
					}
				]
			]
		}
	}
}
```
**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Node Version: v12.14.1

**Additional context**
Lambda environment variables "STORAGE_ZZZZZZ_BUCKETNAME" with "env" value.

It occurres "AccessDeniedException", when use "API_XXXXXX_YYYYYYTABLE_NAME" by AWS.DynamoDB.DocumentClient.

**Comment by kaustavghosh06** [comment]
@chakimar Was able to reproduce this. Marking this as a bug.
The table ARN is correct though right?

**Comment by chakimar** [comment]
> The table ARN is correct though right?
@kaustavghosh06 Yes, "API_XXXXXX_YYYYYYTABLE_ARN" is right.

**Comment by ammarkarachi** [comment]
@chakimar We released a new version 4.17.2 with the fix, don't hesitate to reopen or create another issue
